eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala

```diff
@@ -230,48 +250,43 @@ class RouteResponseSerializer extends CustomSerializerOnly[RouteResponse](_ => {
     JArray(nodeIds.toList.map(n => JString(n.toString)))
 })
 
-class ThrowableSerializer extends CustomSerializerOnly[Throwable](_ => {
+object ThrowableSerializer extends MinimalSerializer({
   case t: Throwable if t.getMessage != null => JString(t.getMessage)
   case t: Throwable => JString(t.getClass.getSimpleName)
 })
 
-class FailureMessageSerializer extends CustomSerializerOnly[FailureMessage](_ => {
+object FailureMessageSerializer extends MinimalSerializer({
   case m: FailureMessage => JString(m.message)
 })
 
-class FailureTypeSerializer extends CustomSerializerOnly[FailureType](_ => {
+object FailureTypeSerializer extends MinimalSerializer({
   case ft: FailureType => JString(ft.toString)
 })
 
-class NodeAddressSerializer extends CustomSerializerOnly[NodeAddress](_ => {
+object NodeAddressSerializer extends MinimalSerializer({
   case n: NodeAddress => JString(HostAndPort.fromParts(n.socketAddress.getHostString, n.socketAddress.getPort).toString)
 })
 
-class DirectedHtlcSerializer extends CustomSerializerOnly[DirectedHtlc](_ => {
-  case h: DirectedHtlc => new JObject(List(("direction", JString(h.direction)), ("add", Extraction.decompose(h.add)(
-    DefaultFormats +
-      new ByteVector32Serializer +
-      new ByteVectorSerializer +
-      new PublicKeySerializer +
-      new MilliSatoshiSerializer +
-      new CltvExpirySerializer))))
-})
+// @formatter:off
+private case class DirectedHtlcJson(direction: String, add: UpdateAddHtlc)
+object DirectedHtlcSerializer extends ConvertClassSerializer[DirectedHtlc](h => DirectedHtlcJson(direction = h.direction, add = h.add))
```

Comment:
Note here how we don't import the lower level serializers. We only need to take care of the actual `DirectedHtlcJson` class.